hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsBackoffMetrics.java

```diff
@@ -0,0 +1,256 @@
+    String metricName = getMetricName(metric, retryValue);
+    switch (metric.getStatisticType()) {
+      case TYPE_COUNTER:
+        return lookupCounterValue(metricName);
+      case TYPE_GAUGE:
+        return lookupGaugeValue(metricName);
+      default:
+        return 0;
+    }
+  }
+
+  /**
+   * Retrieves the value of a specific metric.
+   *
+   * @param metric the metric enum
+   * @return the value of the metric
+   */
+  public long getMetricValue(AbfsBackoffMetricsEnum metric) {
+    return getMetricValue(metric, null);
+  }
+
+  /**
+   * Increments the value of a specific metric.
+   *
+   * @param metric the metric enum
+   * @param retryValue the retry value
+   */
+  public void incrementMetricValue(AbfsBackoffMetricsEnum metric, RetryValue retryValue) {
+    String metricName = getMetricName(metric, retryValue);
+    switch (metric.getStatisticType()) {
+      case TYPE_COUNTER:
+        incCounterValue(metricName);
+        break;
+      case TYPE_GAUGE:
+        incGaugeValue(metricName);
+        break;
+      default:
+        // Do nothing
+        break;
+    }
+  }
+
+  /**
+   * Increments the value of a specific metric.
+   *
+   * @param metric the metric enum
+   */
+  public void incrementMetricValue(AbfsBackoffMetricsEnum metric) {
+    incrementMetricValue(metric, null);
+  }
+
+  /**
+   * Sets the value of a specific metric.
+   *
+   * @param metric the metric enum
+   * @param value the new value of the metric
+   * @param retryValue the retry value
+   */
+  public void setMetricValue(AbfsBackoffMetricsEnum metric, long value, RetryValue retryValue) {
+    String metricName = getMetricName(metric, retryValue);
+    switch (metric.getStatisticType()) {
+      case TYPE_COUNTER:
+        setCounterValue(metricName, value);
+        break;
+      case TYPE_GAUGE:
+        setGaugeValue(metricName, value);
+        break;
+      default:
+        // Do nothing
+        break;
+    }
+  }
+
+  /**
+   * Sets the value of a specific metric.
+   *
+   * @param metric the metric enum
+   * @param value the new value of the metric
+   */
+  public void setMetricValue(AbfsBackoffMetricsEnum metric, long value) {
+    setMetricValue(metric, value, null);
+  }
+
+  /*
+  Acronyms :-
+  1.RCTSI :- Request count that succeeded in x retries
+  2.MMA :- Min Max Average (This refers to the backoff or sleep time between 2 requests)
+  3.s :- seconds
+  4.BWT :- Number of Bandwidth throttled requests
+  5.IT :- Number of IOPS throttled requests
+  6.OT :- Number of Other throttled requests
+  7.NFR :- Number of requests which failed due to network errors
+  8.%RT :- Percentage of requests that are throttled
+  9.TRNR :- Total number of requests which succeeded without retrying
+  10.TRF :- Total number of requests which failed
+  11.TR :- Total number of requests which were made
+  12.MRC :- Max retry count across all requests
+   */
+  @Override
+  public String toString() {
```

Comment:
To improve readability in the toString method, consider breaking down each metric section ($RCTSI$, $MMA$, $RT$, etc.) into separate private methods. This modular approach will simplify the toString method and make each part easier to understand. Something like :- 

@Override
public String toString() {
    if (getMetricValue(TOTAL_NUMBER_OF_REQUESTS) == 0) {
        return "";
    }
    StringBuilder metricString = new StringBuilder();
    
    metricString.append(getRCTSIMetrics());
    metricString.append(getMmaMetrics());
    metricString.append(getThrottlingMetrics());

    return metricString.toString();
}